include/dt-bindings/gpio/gpio.h

```diff
@@ -1,180 +1,453 @@
+#define GPIO_DIR_OUT_HIGH       (GPIO_IO_OUT_ENABLED	\
+				 | GPIO_IO_INIT_ENABLED	\
+				 | GPIO_IO_INIT_HIGH)
 /** @} */
 
 
 /**
- * @name GPIO polarity flags
- * The `GPIO_POL_*` flags are used with `gpio_pin_configure` or `gpio_port_configure`,
- * to specify the polarity of a GPIO pin.
+ * @name GPIO interrupt flags
+ *
+ * The `GPIO_INT_*` flags are used with `gpio_pin_configure` or
+ * `gpio_port_configure`, to specify how input GPIO pins will trigger
+ * interrupts.
+ *
  * @{
  */
 /** @cond INTERNAL_HIDDEN */
-#define GPIO_POL_POS		7
+#define GPIO_INT_SHIFT          9
+#define GPIO_INT_MASK           (0x1F << GPIO_INT_SHIFT)
 /** @endcond */
 
-/** GPIO pin polarity is normal. */
-#define GPIO_POL_NORMAL		(0 << GPIO_POL_POS)
+/** Mask to isolate the `GPIO_INT_LOW` and `GPIO_INT_HIGH` flags. */
+#define GPIO_INT_DIR_MASK       (0x03 << (GPIO_INT_SHIFT + 2))
 
-/** GPIO pin polarity is inverted. */
-#define GPIO_POL_INV		(1 << GPIO_POL_POS)
+/** GPIO pin causes interrupt.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_ENABLE (1 << GPIO_INT_SHIFT)
 
-/** @cond INTERNAL_HIDDEN */
-#define GPIO_POL_MASK		(1 << GPIO_POL_POS)
-/** @endcond */
-/** @} */
+/** Trigger detection when signal is observed to be at a particular
+ * level.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_LEVEL  (0 << (GPIO_INT_SHIFT + 1))
+
+/** Trigger detection when a signal is observed to change level.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_EDGE   (1 << (GPIO_INT_SHIFT + 1))
 
+/** Trigger detection when input state is (or transitions to) low.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_LOW    (1 << (GPIO_INT_SHIFT + 2))
+
+/** Trigger detection on input state is (or transitions to) high.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_HIGH   (1 << (GPIO_INT_SHIFT + 3))
+
+/** Interrupt triggers on both rising and falling edge.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ *
+ * For compatibility with old code this should be combined with
+ * `GPIO_INT_EDGE`.
+ *
+ * For compatibility with new code this should be combined with both
+ * `GPIO_INT_LOW` and `GPIO_INT_HIGH`.  Updated driver code should
+ * prefer to check the individual flags.
+ *
+ * @deprecated Replace tests with a check that flags masked by
+ * `GPIO_INT_DIR_MASK` is equal to `GPIO_INT_LOW | GPIO_INT_HIGH`.
+ */
+#define GPIO_INT_DOUBLE_EDGE    (1 << (GPIO_INT_SHIFT + 4))
```

Comment:
It is deprecated, three lines up.  It's also used as a single-bit flag by every driver that supports double-edge detection.  It can go away when all the drivers have been updated.

Reply:
Clear, though we should probably rephrase or remove the comment
```
 * For compatibility with new code this should be combined with both
 * `GPIO_INT_LOW` and `GPIO_INT_HIGH`.  Updated driver code should
 * prefer to check the individual flags.
```
which seems to suggest that the new code should still use it.